locksmith/src/websub/jobs/keys.ts

```diff
@@ -1,19 +1,42 @@
 import { Key } from '../../graphql/datasource'
-import { Hook } from '../../models'
+import { Hook, ProcessedHookItem } from '../../models'
 import { TOPIC_KEYS } from '../topics'
 
 import { networkMapToFnResult, notifyHook } from '../helpers'
 
+async function getUnprocessedKeys(network: number) {
+  const keySource = new Key(network)
+  const processedKeys = await ProcessedHookItem.findAll({
+    where: {
+      type: 'key',
+      network,
+    },
+    order: [['id', 'DESC']],
+  })
+  const keys = await keySource.getKeys({ first: 25 })
```

Comment:
See my comment above, but I think we should start with this, then check these 25 (let's make that a constant at he top of the file) are _really_ new and only process these.
If the 25 are indeed _really_ new, then get the next 25... etc.
This way I think we cannot "miss" any?

Reply:
Though, this will require changes in the datasource for adding offset. It doesn't support it currently.

Reply:
Ha! I think you can use `skip`: https://thegraph.com/docs/en/developer/graphql-api/

Reply:
Yeah and there is a PR for adding support for that.